src/pcm.c

```diff
@@ -1531,20 +1537,28 @@ static int pcm_mmap_transfer(struct pcm *pcm, void *buffer, unsigned int frames)
     }
 
     while (frames) {
+	pcm_sync_ptr(pcm, 0);
         avail = pcm_mmap_avail(pcm);
 
-        if (!avail) {
+        if (avail < pcm->config.avail_min) {
+            int time = -1;
             if (pcm->flags & PCM_NONBLOCK) {
                 errno = EAGAIN;
                 break;
             }
-
+	    if (pcm->flags & PCM_NOIRQ) {
+                time = (pcm->config.avail_min - avail) / pcm->noirq_frames_per_msec;
+	        /* check the wait time and ensure it's at least 1ms */
+	        if (time < 1)
+		    time = 1;
+	    }
             /* wait for interrupt */
```

Comment:
Could you please also help to update this comment?

Reply:
updated.